roles/test-utils/mining-device/src/main.rs

```diff
@@ -669,3 +674,70 @@ fn generate_random_32_byte_array() -> [u8; 32] {
     rng.fill(&mut arr[..]);
     arr
 }
+
+fn start_mining_threads(
+    have_new_job: Receiver<()>,
+    miner: Arc<Mutex<Miner>>,
+    share_send: Sender<(u32, u32, u32, u32)>,
+) {
+    tokio::task::spawn(async move {
+        let mut killers: Vec<Arc<AtomicBool>> = vec![];
+        loop {
+            let p = std::thread::available_parallelism().unwrap().get() as u32 - 1;
```

Comment:
Noob Questions:
1. In systems that have only a single core without hyper-threading, wouldn't `p`  reduce to 0, potentially leading to undefined behavior? Should we add a check to prevent this?
2. Regarding nonce distribution, we are currently not utilizing the entire nonce space, as `u32::MAX % p ` values are not considered, although the number missed is relatively small, at most `p-1`.

Reply:
1. definitively.
2. I don't see it as an issue.